mainapp/admin.py

```diff
@@ -49,47 +48,49 @@ class VolunteerAdmin(admin.ModelAdmin):
     readonly_fields = ('joined',)
     list_display = ('name', 'phone', 'organisation', 'joined')
     list_filter = ('district', 'joined',)
+
     def download_csv(self, request, queryset):
```

Comment:
Shouldn't we be caching/memoizing the result for this? This probably adds to the performance/uptime issues.
How often is this hit?